### DISCUSSION TITLE
Unable to get forecast

### DISCUSSION THREAD

**Discussion by whereizben** [cytech/Home-Assistant-wundergroundpws#150] [General]
Hi,
I am using 1.10 on the most up to date HASS OS and Core (pasted below) - before and after updating HASS to 2023.1.7 (was using 1.6 previously) I have started get these three errors in my log and unable to retrieve the forecast. The data from my PWS (both indoors and outdoors) comes through fine, but not the forecast. I updated the API key, which did not change anything. Any thoughts?
Thanks!
Ben

Unable to prepare setup for platform wundergroundpws.sensor: Unable to set up component.
15:30:18 – (ERROR) setup.py - message first occurred at 15:30:18 and shows up 2 times
Error during setup of component wundergroundpws
15:30:18 – (ERROR) Wundergroundpws (custom integration)
Check WUnderground API ('NO CURRENT RESULT',)
15:30:18 – (ERROR) Wundergroundpws (custom integration)

Home Assistant 2023.1.7
Supervisor 2022.12.1
Operating System 9.4

**Comment by Malvazar** [comment]
I am also receiving these errors.

**Comment by drjjr2** [comment]
Same. Not sure if it's related to the update or something with wunderground.

**Comment by kwstump** [comment]
Same error here.  Started after an update (Cannot remember if it was a HACS or HA update)

**Reply by kwstump** [reply]
I can confirm it now is working without any changes on my end minus another restart of HA

**Comment by drjjr2** [comment]
I can tell you that direct php requests were failing. I just tried one again and it worked, restarted HA and everything is back.
I'm on 2023.1.7 so it wasn't a problem with HA, just coincidental.

**Comment by cytech** [comment]
i think this has nothing to do with upgrade. api failure f some sort when you are restarting hass.
this error says that the WU current api  returned no result.
at this time unfortunately when encountered during startup the platform fails setup, for, among other reasons, if you have not manually configured latitude and longitude, the component attempts to retrieve it from the API current result (which does not exist).
manually try the api:
`https://api.weather.com/v2/pws/observations/current?stationId=YOURSTATIONID&format=json&units=e&numericPrecision =decimal&apiKey=YOURAPIKEY`
if it returns data, restart hass

**Reply by kwstump** [reply]
I can confirm it now is working without any changes on my end minus another restart of HA.  Thanks cytech for suggesting another restart.

**Comment by whereizben** [comment]
Agreed, it seems to be working now again for me. Thanks!

**Comment by cytech** [comment]
closed as outdated

**Comment by cytech** [comment]
closed as outdated
